catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -59,55 +60,72 @@
 
     private int subDirLevels;
 
+    private PeriodicBatchExecutor<Metacard> periodicDrain;
+
     private boolean enableBackupPlugin = true;
 
     public CatalogBackupPlugin() {
+
+        init();
+    }
+
+    private void init() {
         subDirLevels = 0;
+        setPeriodicDrain(new PeriodicBatchExecutor<>(1000, 60, TimeUnit.SECONDS));
```

Comment:
I'll do that. I think it would look like this:

```
<bean id="executor" class="ddf.catalog.backup.PeriodicBatchExcecutor">
   <argument value="1000"/>      
   <argument value="60"/>      
   <argument value="TimeUnit.SECONDS"/>      
</bean>

 <bean id="catalogBackupPlugin" class="ddf.catalog.backup.CatalogBackupPlugin"
          destroy-method="shutdown">
        <property name="setPeriodicDrain">
	    <ref bean="executor"/>
        </property>
```
Does that look right?